src/foam/u2/wizard/views/PermissiveEditWizardletBorder.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * @license
+ * Copyright 2023 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.wizard.views',
+  name: 'PermissiveEditWizardletBorder',
+  extends: 'foam.u2.wizard.views.NullEditWizardletBorder',
+  documentation: 'Border for wizardlets that the user is allowed to edit',
+
+  imports: ['ctrl'],
+
+  requires: [
+    'foam.u2.borders.CardBorder'
+  ],
+  
+  messages: [
+    {
+      name: 'SAVE_FAILED',
+      message: 'Your information was not saved. Please try again.'
+    }
+  ],
+
+  css: `
+    ^flex{
+      display: flex;
+      justify-content: space-between;
+      align-items: center;
+      // margin: 8px 0;
+    }
+    ^button-flex{
+      display: flex;
+      justify-content: flex-end;
+    }
+ 
+  `,
+
+  properties: [
+    {
+      class: 'FObjectProperty',
+      name: 'oldData'
+    }
+  ],
+  
+  methods: [
+    function init() {
+      // set in init as factories are lazy
+      this.wizardlet.loadEvent.sub(() => {
+        this.oldData = this.wizardlet.data.clone()
+      })
+
+      this
+        .addClass()
+          .startContext({data: this})
+            .start().addClass(this.myClass('flex'))
+            .start().addClass('h500', this.myClass('wizard-heading')).add(this.title).end()
+            .start().addClass(this.myClass('button-flex'))
+              .add(this.EDIT, ' ', this.SAVE, ' ', this.CANCEL)
+            .end()
+            .end()
+          .endContext()
+          .start(this.CardBorder).addClass(this.myClass('card'))
+          .tag('div', null, this.content$)
+        .end()
+    }
+  ],
+
+  actions: [
+    {
+      name: 'Edit',
+      label: 'Edit',
+      buttonStyle: 'LINK',
+      isAvailable: function(controllerMode) {
+        return this.controllerMode !== foam.u2.ControllerMode.EDIT
+      },
+      code: function() {
+        this.controllerMode = foam.u2.ControllerMode.EDIT
+      }
+    },
+    {
+      name: 'Save',
+      label: 'Save',
+      buttonStyle: 'LINK',
+      isAvailable: function(controllerMode) {
+        return this.controllerMode === foam.u2.ControllerMode.EDIT
+      },
+      code: async function () {      
+       try {
+        await this.wizardlet.save()
+        this.ctrl.reloadClient()
+        this.controllerMode = foam.u2.ControllerMode.VIEW
+       } catch (e) {
+        alert(this.SAVE_FAILED)
```

Comment:
Is this the default javascript 'alert' or does it call/use our notify? 
if using notify we can add the e.message as the description, if it conveys info on perhaps a field/property that is invalid.